cloud/scope/machine.go

```diff
@@ -87,6 +88,39 @@ type MachineScope struct {
 	AzureMachine *infrav1.AzureMachine
 }
 
+// VMSpecs returns the VM specs.
+func (m *MachineScope) VMSpecs(ctx context.Context) ([]azure.VMSpec, error) {
```

Comment:
I don't know how I feel about needing context and error as return here... Had to keep them to port over ssh decode/VM image parsing the version and getting the bootstrap data from the secret but not sure that's the best way to handle it yet.

Reply:
So I had to move it because bootstrap data isn't available during delete, so had to be in create only.